python/dgl/graphbolt/itemset.py

```diff
@@ -442,3 +443,188 @@ def __repr__(self) -> str:
             itemsets=itemsets_str,
             names=self._names,
         )
+
+
+class ItemSet4(Dataset):
```

Comment:
Is it just removing `__iter__` from existing `ItemSet/Dict` and inherit from `Dataset` only? Other code are directly copied? If yes, could we just modify upon the existing one directly? will it break existing `ItemSampler`?

Reply:
It's not just simply copying existing `ItemSet`. But I can try to modify upon the existing code directly.